src/Compute/Compute/Generated/Disk/Config/NewAzureDiskPurchasePlanConfig.cs

```diff
@@ -0,0 +1,76 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Compute.Automation.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Compute.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+
+namespace Microsoft.Azure.Commands.Compute.Generated.Disk.Config
+{
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DiskPurchasePlanConfig", DefaultParameterSetName = "DefaultParameterSet")]
+    [OutputType(typeof(PSPurchasePlan))]
+    public class NewAzureDiskPurchasePlanConfig : Microsoft.Azure.Commands.ResourceManager.Common.AzureRMCmdlet
+    {
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true)]
+        public string Publisher { get; set; }
```

Comment:
Why all parameters are optional in the New-* cmdlet. Please raise a design doc for this change: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues